packages/client/src/controllers/engine.ts

```diff
@@ -702,8 +699,9 @@ export class Engine extends IEngine {
 
   private onSessionRequest: EnginePrivate["onSessionRequest"] = (topic, payload) => {
     try {
-      this.isValidRequest({ topic, ...payload.params });
-      this.client.events.emit("request", { ...payload, params: { ...payload.params, topic } });
+      const { id, params } = payload;
+      this.isValidRequest({ topic, ...params });
+      this.client.events.emit("request", { id, topic, params });
```

Comment:
This is also so much nicer in terms of syntax with the explicit object keys 👌 

The `...payload` spreads made it hard to know exactly what the final object looks like without digging into type defs